zeebe/restore/src/main/java/io/camunda/zeebe/restore/RestoreManager.java

```diff
@@ -58,41 +60,44 @@ public RestoreManager(
     this.meterRegistry = meterRegistry;
   }
 
-  public CompletableFuture<Void> restore(final long backupId, final boolean validateConfig) {
-    final Path dataDirectory = Path.of(configuration.getData().getDirectory());
-    try {
-      if (!dataFolderIsEmpty(dataDirectory)) {
-        LOG.error(
-            "Brokers's data directory {} is not empty. Aborting restore to avoid overwriting data. Please restart with a clean directory.",
-            dataDirectory);
-        return CompletableFuture.failedFuture(
-            new DirectoryNotEmptyException(dataDirectory.toString()));
-      }
-    } catch (final IOException e) {
-      return CompletableFuture.failedFuture(e);
+  public void restore(final long backupId, final boolean validateConfig)
+      throws IOException, ExecutionException, InterruptedException {
+    final var dataDirectory = Path.of(configuration.getData().getDirectory());
+    if (!dataFolderIsEmpty(dataDirectory)) {
+      LOG.error(
+          "Brokers's data directory {} is not empty. Aborting restore to avoid overwriting data. Please restart with a clean directory.",
+          dataDirectory);
+      throw new DirectoryNotEmptyException(dataDirectory.toString());
     }
 
     final var partitionToRestore = collectPartitions();
 
```

Comment:
Nitpick, make it plural for clarity

```suggestion
    final var partitionsToRestore = collectPartitions();
```